pkg/gen/nodejs-templates/yaml.ts.mustache

```diff
@@ -235,20 +251,36 @@ import * as outputApi from "../types/output";
             super("kubernetes:yaml:ConfigFile", name, config, opts);
             const fileId = config && config.file || name;
             let text: Promise<string>;
-            if (fileId.startsWith("http://") || fileId.startsWith("https://")) {
-                text = fetch(fileId).then(r => r.text())
+            if (isUrl(fileId)) {
+                text = fetch(fileId).then(r => {
+                    if (r.ok) {
+                        return r.text()
+                    } else {
+                        throw Error(`Error fetching YAML file '${fileId}': ${r.status} ${r.statusText}`);
+                    }
+                });
             } else {
                 text = Promise.resolve(fs.readFileSync(fileId).toString());
             }
 
-            this.resources = pulumi.output(text.then(t => parseYamlDocument({
-                objs: jsyaml.safeLoadAll(t),
-                transformations: config && config.transformations || [],
-                resourcePrefix: config && config.resourcePrefix || undefined
-            }, {parent: this})));
+            this.resources = pulumi.output(text.then(t => {
+                try {
+                    return parseYamlDocument({
+                        objs: yamlLoadAll(t),
+                        transformations: config && config.transformations || [],
+                        resourcePrefix: config && config.resourcePrefix || undefined
+                    }, {parent: this})
+                } catch (e) {
+                    throw Error(`Error fetching YAML file '${fileId}': ${e}`);
+                }
+            }));
         }
     }
 
+    /** @ignore */ function isUrl(s: string): boolean {
+        return s.startsWith("http://") || s.startsWith("https://")
```

Comment:
Should we be instead attempt to parse a URL with https://nodejs.org/api/url.html#url_class_url and return `true` if parsing succeeds? This would allow us to handle file URLs, etc.

Reply:
(I do see that what you've written matches the prior behavior of `ConfigFile`, so no urgency here)